osh/glob_.py

```diff
@@ -491,17 +557,28 @@ def _Glob(self, arg, out):
                 results = tmp  # idiom to work around mycpp limitation
                 n = len(results)
```

Comment:
Hm OK I see we respect

- dotglob and GLOBIGNORE
- no_dash_glob
- then there's GLOBIGNORE
- then we don't fully support shopt -s globskipdots

But I see why you did that!  This code is complex, but that's just how bash is  ...